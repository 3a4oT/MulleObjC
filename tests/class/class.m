#import <MulleStandaloneObjC/MulleStandaloneObjC.h>


@interface Foo : NSObject
@end


@implementation Foo

+ (Class) class
{
   Class  cls;

   cls = [super class];
   printf( "%s: %s\n", __PRETTY_FUNCTION__, cls == mulle_objc_object_get_class( self) ? "SAME" : "DIFFERENT");
   return( cls);
}


- (Class) class
{
   Class  cls;

   cls = [super class];
   printf( "%s: %s\n", __PRETTY_FUNCTION__, cls == mulle_objc_object_get_class( self) ? "SAME" : "DIFFERENT");
   return( cls);
}

@end


main()
{
   [Foo class];
   [[Foo new] class];
}
