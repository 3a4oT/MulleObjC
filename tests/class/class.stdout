+[Foo class]: SAME
-[Foo class]: SAME
